compiler/lib/driver.ml

```diff
@@ -147,65 +141,52 @@ let print p =
   p
 
 let rec loop max name round i (p : 'a) : 'a =
+  if times () then Format.eprintf "%s#%d...@." name i;
   let p' = round p in
-  if i >= max || Code.eq p' p
-  then p'
-  else (
-    if times () then Format.eprintf "Start Iteration (%s) %d...@." name i;
-    loop max name round (i + 1) p')
-
-let identity x = x
+  if i >= max
+  then (
+    if times () then Format.eprintf "%s#%d: couldn't reach fix point.@." name i;
+    p')
+  else if Code.eq p' p
+  then (
+    if times () then Format.eprintf "%s#%d: fix-point reached.@." name i;
+    p')
+  else loop max name round (i + 1) p'
 
 (* o1 *)
 
 let o1 : 'a -> 'a =
   print
   +> tailcall
-  +> flow_simple (* flow simple to keep information for future tailcall opt *)
-  +> specialize'
+  +> flow
+  +> specialize
   +> eval
   +> inline (* inlining may reveal new tailcall opt *)
   +> deadcode
   +> tailcall
   +> phi
   +> flow
-  +> specialize'
+  +> specialize
   +> eval
   +> inline
   +> deadcode
   +> print
```

Comment:
I'm wondering why there is a `print` here.

Reply:
history -> f28eb2b6fdf6395ec620e60ee077cb7b99bd35eb